.travis.yml

```diff
@@ -1,48 +1,58 @@
 language: php
-php:
-  - 7.1
-  - 7.2
-  - nightly
+php: [ 7.1, 7.2, nightly ]
 sudo: false
 
 env:
+
 matrix:
   fast_finish: true
   allow_failures:
     - php: nightly
 
 install:
   - composer install --no-interaction --prefer-dist --optimize-autoloader
+install:
+  - travis_retry composer install --no-interaction --prefer-dist --optimize-autoloader
+  - travis_retry composer global require phpunit/phpunit ^6 # cannot use phpunit.phar or require-dev, because this package is a phpunit dep
+  - travis_retry wget --no-verbose https://phar.io/releases/phive.phar
+
+script:
+  - /home/travis/.composer/vendor/bin/phpunit --no-coverage
+  - bin/behat -p default
 
 jobs:
   include:
-    - stage: test
+    - stage: dogfood
+      php: 7.1
       script:
-      - bin/phpunit --no-coverage
-      - bin/behat -p default
+        - chmod u+x ./bin/phpdoc
+        - ./bin/phpdoc run
 
-    - stage: test
+    - stage: phar
       php: 7.1
       script:
         - php ./bin/console --env=prod cache:warm
         - php -d phar.readonly=false tools/box.phar compile
         - bin/behat -p phar
+        - php build/phpDocumentor.phar run
 
     - stage: coverage
       php: 7.1
       script:
-        - bin/phpunit
+        - /home/travis/.composer/vendor/bin/phpunit
       after_script:
-        - wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover build/logs/clover.xml
-        - wget https://github.com/satooshi/php-coveralls/releases/download/v1.0.1/coveralls.phar && php coveralls.phar --verbose
+        - travis_retry php phive.phar --no-progress install --trust-gpg-keys E82B2FB314E9906E php-coveralls/php-coveralls && ./tools/php-coveralls --verbose
+        - travis_retry wget --no-verbose https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover build/logs/clover.xml
 
     - stage: lint
       php: 7.1
       services:
         - docker
+      before_script:
+        - travis_retry php phive.phar --no-progress install --trust-gpg-keys 8E730BA25823D8B5 phpstan
       script:
+        - ./tools/phpstan analyse src --level max --configuration phpstan.neon
```

Comment:
We are running phpstan at level 2 at the moment. please set that here, and create a new PR to upgrade to max.